File: 2019 maj/python/3.py
with open("liczby.txt", "r") as f:
    liczby = list(map(int, f.read().strip().split('\n')))

def nwd(a, b):
    if b == 0:
        return a
    return nwd(b, a % b)

ans = [0, 0, 0] # pierwsza liczba, długość ciągu, nwd
for i in range(len(liczby)):
    cur, j = [liczby[i], 1, liczby[i]], i + 1
    while j < len(liczby) and nwd(cur[2], liczby[j]) != 1:
        cur[1] += 1
        cur[2] = nwd(cur[2], liczby[j])
        j += 1
    
    if cur[1] > ans[1]:
        ans = cur

print(*ans)

File: 2019 maj/c++/3.cpp
#include <iostream>
#include <fstream>
using namespace std;

int a[505];

struct ciag
{
    int first;
    int len;
    int gcd;
};

int nwd(int a, int b)
{
    if (b == 0)
        return a;
    return nwd(b, a % b);
}

int main()
{
    ifstream in("liczby.txt");
    
    for (int i = 1; i <= 500; i++)
        in >> a[i];
    
    ciag ans = {0, 0, 0};
    for (int i = 1; i <= 500; i++)
    {
        ciag cur = {a[i], 1, a[i]};
        int j = i + 1;
        while (j <= 500 && nwd(cur.gcd, a[j]) != 1)
        {
            cur.gcd = nwd(cur.gcd, a[j]);
            cur.len++;
            j++;
        }
        if (cur.len > ans.len)
            ans = cur;
    }

    cout << ans.first << ' ' << ans.len << ' ' << ans.gcd << '\n';
}